mono/metadata/class.c

```diff
@@ -5692,13 +5693,21 @@ mono_class_setup_supertypes (MonoClass *klass)
 	supertypes = mono_class_alloc0 (klass, sizeof (MonoClass *) * ms);
 
 	if (klass->parent) {
-		supertypes [klass->idepth - 1] = klass;
+		CHECKED_METADATA_WRITE_PTR ( supertypes [klass->idepth - 1] , klass );
+
+		// Reference checking is done one assignment at a time, so memcpy cannot be used in checked mode
+#ifdef CHECKED_BUILD
```

Comment:
We should shift as much complexity as possible to checked build itself.

This one-off configuration changes eventually pile up and make code too hard to maintain.

Checked build should not introduce additional complication into the runtime code in the form of extra conditionals.